`tracing-opentelemetry 0.17.1` does not compile with `default-features = false`

## Bug Report
<!--
Thank you for reporting an issue.

Please fill in as much of the template below as you're able.
-->

### Version

`tracing 0.1`
`tracing-opentelemetry 0.17.1`
`opentelemetry 0.17`

<!--
List the versions of all `tracing` crates you are using. The easiest way to get
this information is using `cargo-tree`.

`cargo install cargo-tree`
(see install here: https://github.com/sfackler/cargo-tree)

Then:

`cargo tree | grep tracing`
-->

### Platform

`Linux lungmen 5.16.7 #1-NixOS SMP PREEMPT Sat Feb 5 18:22:06 UTC 2022 x86_64 GNU/Linux`

<!---
Output of `uname -a` (UNIX), or version and 32 or 64-bit (Windows)
-->

### Crates

`tracing-opentelemetry` is affected.

<!--
If known, please specify the affected tracing crates. Otherwise, delete this
section.
-->

### Description

`tracing-opentelemetry 0.17.1` with `default-features = false` fails to compile with:
```
error[E0282]: type annotations needed for `Option<T>`
   --> /home/patriot/.cargo/registry/src/github.com-1ecc6299db9ec823/tracing-opentelemetry-0.17.1/src/layer.rs:590:25
    |
588 |                     let normalized_meta = None;
    |                         --------------- consider giving `normalized_meta` the explicit type `Option<T>`, where the type parameter `T` is specified
589 |                     let (file, module) = match &normalized_meta {
590 |                         Some(meta) => (
    |                         ^^^^^^^^^^ cannot infer type for type parameter `T` declared on the enum `Option`
    |
    = note: type must be known at this point
```

--- a/.github/workflows/check_features.yml
+++ b/.github/workflows/check_features.yml
@@ -44,6 +44,7 @@ jobs:
         - tracing-macros
         - tracing-serde
         - tracing-tower
+        - tracing-opentelemetry
         # tracing and tracing-subscriber have too many features to be checked by
         # cargo-hack --feature-powerset, combinatorics there is exploding.
         #- tracing

--- a/tracing-opentelemetry/src/subscriber.rs
+++ b/tracing-opentelemetry/src/subscriber.rs
@@ -3,8 +3,6 @@ use opentelemetry::{
     trace::{self as otel, noop, TraceContextExt},
     Context as OtelContext, Key, KeyValue, Value,
 };
-#[cfg(not(feature = "tracing-log"))]
-use std::borrow::Cow;
 use std::fmt;
 use std::marker;
 use std::time::{Instant, SystemTime};

--- a/tracing-opentelemetry/src/subscriber.rs
+++ b/tracing-opentelemetry/src/subscriber.rs
@@ -585,7 +585,7 @@ where
                     let builder_attrs = builder.attributes.get_or_insert(Vec::new());
 
                     #[cfg(not(feature = "tracing-log"))]
-                    let normalized_meta = None;
+                    let normalized_meta: Option<tracing_core::Metadata<'_>> = None;
                     let (file, module) = match &normalized_meta {
                         Some(meta) => (
                             meta.file().map(|s| Value::from(s.to_owned())),

